packages/cli-e2e/__tests__/angular.specs.ts

```diff
@@ -65,20 +66,19 @@ describe('ui', () => {
         }),
       ]);
 
-      const startServerProcess = processManager.spawn('npm', ['run', 'start'], {
-        cwd: getProjectPath(projectName),
-      });
+      const terminalServer = new Terminal(processManager, [
+        'npm',
+        ['run', 'start'],
+        {
+          cwd: getProjectPath(projectName),
+        },
+      ]);
 
-      await new Promise<void>((resolve) => {
-        startServerProcess.stdout.on('data', async (data) => {
-          if (
-            /Compiled successfully/.test(
-              stripAnsi(data.toString()).replace(/\n/g, '')
-            )
-          ) {
-            resolve();
-          }
-        });
+      await new Promise((resolve) => {
+        terminalServer
```

Comment:
The API is pretty clean/straightforward to understand 💯 

Proposition (feel free to reject):

Since the condition is always going to be performed `.on` a stream (stdout, stderr, unless I'm mistaken),
maybe we could just combine that and the `.on` becomes superfluous.

eg:

```
terminalserver
  .when('stdout', /Compiled successfully/)
  .do(resolve)
```

Reply:
> The API is pretty clean/straightforward to understand 💯

Thanks! Hopefully this will ease the E2E testing process :D 

About your proposition, I thought about it and this is a conscious choice:
 - I think even if it's a tad bit more verbose, having one args per chaining func allow to have a verbose and clear understanding of what's going on.
 - A Condition can also be applied 'on metal', directly on the process.
 
 
 However, I agree that `on` will very often be `on('stdout')`. Which make it tedious. My plan in the future is that the grammar of the toolbelt (i.e. what can follow what in the func chaing) to be way more flexible.
Meaning that we'll have 'defaulting' on some part of the grammar: which will make this possible:

```typescript
// Will be the same...
terminalserver
  .when(/Compiled successfully/)
  .do(resolve);
// As this:
terminalserver
  .when(/Compiled successfully/)
  .on('stdout')
  .do(resolve);
```